packages/components/text/src/text.tsx

```diff
@@ -0,0 +1,284 @@
+import type { MessageDescriptor } from 'react-intl';
+
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import { useTheme } from '@emotion/react';
+import invariant from 'tiny-invariant';
+import { oneLine } from 'common-tags';
+import {
+  filterDataAttributes,
+  warnDeprecatedProp,
+} from '@commercetools-uikit/utils';
+import {
+  bodyStyles,
+  detailStyles,
+  headlineStyles,
+  subheadlineStyles,
+  wrapStyles,
+} from './text.styles';
+
+type TBasicTextProps = {
+  intlMessage?: MessageDescriptor;
+  children?: React.ReactNode;
+};
+
+type TBasicHeadlineProps = {
+  title?: string;
+};
+
+const getIsIntlMessage = (
+  intlMessage: unknown
+): intlMessage is MessageDescriptor => {
+  if (intlMessage && typeof intlMessage === 'object') {
+    return (
+      'id' in intlMessage &&
+      'description' in intlMessage &&
+      'defaultMessage' in intlMessage
+    );
+  }
+  return false;
+};
+
+const warnIfMissingTitle = (
+  props: TBasicHeadlineProps,
+  componentName: string
+) => {
+  if (typeof props.title === 'string') {
+    invariant(
+      props.title.length > 0,
+      `Invalid prop 'title' supplied to '${componentName}'. Expected it to be nonempty string, but it was empty.`
+    );
+  }
+};
+
+const warnIfMissingContent = (
+  props: TBasicTextProps,
+  componentName: string
+) => {
+  const hasContent =
+    getIsIntlMessage(props.intlMessage) ||
```

Comment:
Hmm I don't think we need this. Is there a particular reason for that? 
Here we only check if the `prop.intlMessage` is defined. The format of the prop is then checked by the TS type.

Reply:
I want to make sure that it is a `MesssageDescriptor` and a string as a result of `intl.formatMessage` from the consumer side.

Reply:
Yes TS checks that. And if the user passes a wrong object, react-intl complains. I don't see a point in having to check things extra on our own.

Reply:
I get that, but note that the consumer won't be fully aware the `intlMessage=MessageDescriptor` type.
Not even intellisense helps here

(example from a plain js consumer side)
![image](https://user-images.githubusercontent.com/462507/106126762-1742b580-615e-11eb-8153-52f914378fef.png)

Reply:
Consider that JS users still have the prop type, which should be an object if I'm not mistaken (you can inspect the bundles).

It's a little trade off at the end.